packit_service/worker/psbugzilla.py

```diff
@@ -0,0 +1,163 @@
+
+        self.logger.info(f"Creating a new bug for {component} in {product}:{version}")
+        try:
+            newbug = self.api.createbug(createinfo)
+        except Fault as exc:
+            msg = f"Failed to create a bug with {createinfo}. Exception: {exc.faultString}"
+            raise RuntimeError(msg)
+        self.logger.info(f"Created bug #{newbug.id} at {newbug.weburl}")
+        return newbug.id, newbug.weburl
+
+    @backoff.on_exception(wait_gen=backoff.expo, exception=RuntimeError, max_time=30)
+    def add_patch(self, bzid: int, content: bytes, file_name: str = None) -> int:
+        """
+        Add 'content' as attachment/patch into bug 'bzid'.
+
+        :param bzid: Bugzilla bug to add patch to
+        :param content: patch content
+        :param file_name: attachment/patch file name
+        :return: attachment id
+        """
+        with TemporaryFile() as tmp_file:
+            tmp_file.write(content)
+            tmp_file.flush()
+            tmp_file.seek(0)
+            self.logger.info(f"Adding an attachment to bug #{bzid}")
+            try:
+                attachment_id = self.api.attachfile(
+                    idlist=bzid,
+                    attachfile=tmp_file,
+                    description="Approved patch",
+                    file_name=file_name or "patch",
+                    is_patch=True,
+                    content_type="text/plain",
+                )
+            except Fault as exc:
+                # This might be a 'query serialization error' if the bug has just been created.
+                # Hence the @backoff.on_exception to retry.
+                msg = f"Failed to add an attachment to bug #{bzid}. Exception: {exc.faultString}"
+                raise RuntimeError(msg)
+            self.logger.info(f"Added attachment {attachment_id} to bug #{bzid}")
+        return attachment_id
+
+
+def get_pr(namespace: str, repo: str, pr_num: int) -> PullRequest:
+    service = PagureService(
+        instance_url=getenv("PAGURE_URL") or "https://git.stg.centos.org",
+        token=getenv("PAGURE_TOKEN"),
+        read_only=True,
+    )
+    project = service.get_project(namespace=namespace, repo=repo)
+    return project.get_pr(pr_num)
+
+
+# PAGURE_TOKEN="ABC" BUGZILLA_API_KEY="XYZ" python3 psbugzilla.py
+if __name__ == "__main__":
+    repo = getenv("REPO") or "rpm"
+    pr_id = int(getenv("PR_ID")) or 19
+    pr = get_pr(getenv("NAMESPACE") or "source-git", repo, pr_id)
+    bz = Bugzilla()
+    logging.basicConfig()
+    bz.logger.setLevel(logging.DEBUG)
+    description = f"Based on approved CentOS Stream Pull Request: {pr.url}"
+    bzid, url = bz.create_bug(
+        product="Red Hat Enterprise Linux 8",
+        version="8.2",
+        component=repo,
+        summary=f"[CentOS Stream] {pr.title}",
+        description=description,
+    )
+    bz.add_patch(bzid=bzid, content=pr.patch, file_name=f"centos-pr-{pr_id}.patch")
```

Comment:
this is nice, I'm assuming that in the next iteration this would be a dedicated function

Reply:
and we could actually have another function to pick things from dist-git